doc/setup_instructions.md

```diff
@@ -521,13 +521,18 @@ the pre-commit hook section for how to speed things up.
 Pre-commit hook
 ---------------
 
-On Linux you can enable a pre-commit hook to automatically run the
+You can enable a pre-commit hook to automatically run the
 formatting checks before each commit to avoid accidentally committing
-code with formatting issues. To install the hook run the following
-script:
+code with formatting issues.
+
+To install pre-commit use the commands for Windows `pip install pre-commit`,
+if you are using Linux you can install with your package manager.
+More installation instruction can be seen [here](https://pre-commit.com/#installation)
+
+To finally install the hook run in terminal:
```

Comment:
I think the text would flow a bit better if this was `Then, to install the hook...`
also terminal should not be mentioned directly, it should be either said that just run the following command *or* explicitly mention command prompt and terminal.

Reply:
What do you think about this?
```
To install pre-commit on Windows run `pip install pre-commit`,
If on Linux run `sudo pip install pre-commit`

Then, to install the hook run `pre-commit install`
```

Reply:
I think it shouldn't be split platform specific like that. Instead there should be a note saying on linux, you can use sudo pip install..

Reply:
how about this?
```
To install pre-commit run `pip install pre-commit`, on Linux you would need to run with `sudo`
```

then probably add this at the end or troubleshooting part?

```
More installation instruction can be seen [here](https://pre-commit.com/#installation)
```

Reply:
The thing, is you don't need to run with sudo. You can do what is required before. I think that it should be given as an additional tip saying that you *can* run with sudo, but also the previously shown approach with `--user` flag for pip should work just fine as well.